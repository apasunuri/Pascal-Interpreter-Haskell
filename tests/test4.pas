program test4;

var
    a, b: real;

begin 
    a := 10;
    while a < 20 do
        begin
            writeln('value of a:');
            writeln(a);
            a := a + 1;
        end;
    writeln();
    for b := 1 to 8 do
        begin
            writeln('SIN, COS, EXP, LN of b:');
            writeln(sin(b));
            writeln(cos(b));
            writeln(exp(b));
            writeln(ln(b));
            writeln();
        end; 
end.