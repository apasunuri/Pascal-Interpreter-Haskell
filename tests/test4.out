value of a:
10.0       
value of a:
11.0       
value of a:
12.0
value of a:
13.0
value of a:
14.0
value of a:
15.0
value of a:
16.0
value of a:
17.0
value of a:
18.0
value of a:
19.0

SIN, COS, EXP, LN of b:
0.84147096
0.5403023
2.7182817
0.0

SIN, COS, EXP, LN of b:
0.9092974
-0.41614684
7.389056
0.6931472

SIN, COS, EXP, LN of b:
0.14112
-0.9899925
20.085537
1.0986123

SIN, COS, EXP, LN of b:
-0.7568025
-0.6536436
54.59815
1.3862944

SIN, COS, EXP, LN of b:
-0.9589243
0.2836622
148.41316
1.609438

SIN, COS, EXP, LN of b:
-0.2794155
0.96017027
403.4288
1.7917595

SIN, COS, EXP, LN of b:
0.6569866
0.75390226
1096.6332
1.9459101

SIN, COS, EXP, LN of b:
0.98935825
-0.14550003
2980.958
2.0794415